internal/dispatch/dispatch.go

```diff
@@ -37,6 +37,7 @@ type Dispatcher interface {
 // Check interface describes just the methods required to dispatch check requests.
 type Check interface {
 	// DispatchCheck submits a single check request and returns its result.
+	// The result should be safe to access from multiple goroutines.
```

Comment:
FYI i believe this should hold, but right now, I've no clue if it does, my gut tells me it doesn't. And if my gut is right, there could be _lots_ of places where we'd need to shove a `.cloneVT()` call 😬